ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+                                   ",");
+                            nBytes += 1;
+                        }
+                        else
+                        {
+                            strcat(phMiraMonLayer->szStringToOperate + nBytes,
+                                   "]``)");
+                            nBytes += 4;
+                        }
+                    }
+                    poFeature->SetField(nIField,
+                                        phMiraMonLayer->szStringToOperate);
+                }
+                else
+                {
+                    for (nIRecord = 0;
+                         nIRecord <
+                         phMiraMonLayer->pMultRecordIndex[nIElem].nMR;
+                         nIRecord++)
+                    {
+                        GoToFieldOfMultipleRecord(nIElem, nIRecord, nIField);
+                        memset(phMiraMonLayer->szStringToOperate, 0,
+                               phMiraMonLayer->pMMBDXP->pField[nIField]
+                                   .BytesPerField);
+                        fread_function(phMiraMonLayer->szStringToOperate,
+                                       phMiraMonLayer->pMMBDXP->pField[nIField]
+                                           .BytesPerField,
+                                       1, phMiraMonLayer->pMMBDXP->pfDataBase);
+                        phMiraMonLayer
+                            ->szStringToOperate[phMiraMonLayer->pMMBDXP
+                                                    ->pField[nIField]
+                                                    .BytesPerField] = '\0';
+                        MM_RemoveWhitespacesFromEndOfString(
+                            phMiraMonLayer->szStringToOperate);
+
+                        if (phMiraMonLayer->pMMBDXP->CharSet ==
+                            MM_JOC_CARAC_OEM850_DBASE)
+                            MM_oemansi_n(
+                                phMiraMonLayer->szStringToOperate,
+                                phMiraMonLayer->pMMBDXP->pField[nIField]
+                                    .BytesPerField);
+
+                        if (phMiraMonLayer->pMMBDXP->CharSet !=
+                            MM_JOC_CARAC_UTF8_DBF)
+                        {
+                            // MiraMon encoding is ISO 8859-1 (Latin1) -> Recode to UTF-8
+                            char *pszString =
+                                CPLRecode(phMiraMonLayer->szStringToOperate,
+                                          CPL_ENC_ISO8859_1, CPL_ENC_UTF8);
+
+                            CPLStrlcpy(
+                                phMiraMonLayer->szStringToOperate, pszString,
+                                (size_t)phMiraMonLayer->pMMBDXP->pField[nIField]
+                                        .BytesPerField +
+                                    1);
+
+                            CPLFree(pszString);
+                        }
+                        papszValues[nIRecord] =
```

Comment:
to avoid memory leaks when reading several features.  But it would be much safer to *not* have a papszValues class member, and just use a local CPLStringList C++ object, that will automatically free memory when the scope terminates
```suggestion
                        CPLFree(papszValues[nIRecord]);
                        papszValues[nIRecord] =
```

Reply:
I think you already did it.